### PR TITLE
Bugzilla 1149666

### PR DESCRIPTION
**Pull request by trink**

### CODE REVIEW

**Comment by mreid-moz** [comment]
Can you rebase this off `master`?

**Comment by trink** [comment]
Why only the last diff is different?

**Comment by mreid-moz** [comment]
To get rid of the duplicated commits from #55.

File: heka/sandbox/encoders/combine_telemetry_objects.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            local ok, object = pcall(cjson.decode, msg.Fields[i].value[1])
+            if ok then
+                json[section][name] = object
+            end
+        end
+    end
+    local ok, payload = pcall(cjson.encode, json)
+    if not ok then return -1, payload end
+
+    inject_payload("txt", "output", msg.clientId, "\t", payload, "\n")
```
Comment (mreid-moz) [inline heka/sandbox/encoders/combine_telemetry_objects.lua:32]: The previous data export included the leftover Fields in the output - there are a few things that are not inside the payload that could be useful, though I'm not sure whether they are neccessary or not. `geoCountry` and `Timestamp` are likely to be of interest.

**Comment by trink** [comment]
It is only a github display issue.

File: heka/sandbox/encoders/combine_telemetry_objects.lua (hunk 1 above)
Comment (trink) [inline heka/sandbox/encoders/combine_telemetry_objects.lua:32]: The goal is to pass back the original structure, correct?

**Comment by mreid-moz** [comment]
I get an error of "error: Failed serializing: bad argument #0 to '?' (must have one string argument)" when trying to use this with the S3 input - is that due to the "raw" message not being properly assigned in the input code?

File: heka/sandbox/encoders/combine_telemetry_objects.lua (hunk 1 above)
Comment (mreid-moz) [inline heka/sandbox/encoders/combine_telemetry_objects.lua:32]: I would like to provide everything in the heka message, with the payload combined back into its original structure.

**Comment by trink** [comment]
Yeah, this PR pulls in the Heka with the fix (i.e. you need to build the pipeline with this tag for this encoder to work)

**Pull request merged**